### PR TITLE
Add workflow that syncs documentation to the website repository on pushes to main

### PR DESCRIPTION
**Pull request by jdbaldry**
We can test and evaluate the Continuous Delivery of documentation to the website repository without publishing it on the website thanks to https://github.com/grafana/website/pull/20393/files which tells Hugo not to render or list any pages under content/docs/explore-logs/.

We can also put in place a similar workflow for publishing versioned releases of documentation after we decide on how those are going to be maintained. `grafana/grafana` uses long-lived version branches for each version and documentation is synced from those branches. `grafana/k6-docs` doesn't need to version documentation alongside the code (because it's a separate repository) and maintains all of its documentation in the `main` branch.

### CODE REVIEW

File: .github/workflows/publish-technical-documentation-next.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    paths:
+      - "docs/sources/**"
+  workflow_dispatch:
+jobs:
+  sync:
+    if: github.repository == 'grafana/explore-logs'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
```
Comment (svennergr) [inline .github/workflows/publish-technical-documentation-next.yml:16]:
```suggestion
```

File: .github/workflows/publish-technical-documentation-next.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        # It must be regenerated in the grafanabot GitHub account and requires a Grafana organization
+        # GitHub administrator to update the organization secret.
+        # The IT helpdesk can update the organization secret.
+        run: git clone --single-branch --no-tags --depth 1 -b master https://grafanabot:${{ secrets.WEBSITE_SYNC_TOKEN }}@github.com/grafana/website-sync ./.github/actions/website-sync
+
+      - name: Publish to website repository (next)
+        uses: ./.github/actions/website-sync
+        id: publish-next
+        with:
+          repository: grafana/website
```
Comment (svennergr) [inline .github/workflows/publish-technical-documentation-next.yml:28]:
```suggestion
          repository: grafana/explore-logs
```

shouldn't it?

**Review by svennergr — COMMENTED** [review]
Guess that needs to change? Not super sure.

File: .github/workflows/publish-technical-documentation-next.yml (hunk 2 above)
Comment (svennergr) [reply .github/workflows/publish-technical-documentation-next.yml:28]: Not sure. How does the action get's the content of `/docs/sources` repo?

Comment (svennergr) [reply .github/workflows/publish-technical-documentation-next.yml:28]: Ok, tried to understand the action a bit more. I think I got it now. Disregard my comments.

**Review by svennergr — APPROVED** [review]

File: .github/workflows/publish-technical-documentation-next.yml (hunk 2 above)
Comment (jdbaldry) [reply .github/workflows/publish-technical-documentation-next.yml:28]: Thanks for taking the time to understand the workflow :)

**Review by matryer — APPROVED** [review]
LGTM - going to try it out

**Pull request merged**